src/Tasks/MigrateContentToElement.php

```diff
@@ -17,26 +54,113 @@ class MigrateContentToElement extends BuildTask
 
     public function run($request)
     {
-        // TODO: needs rewriting for multiple elemental areas
         $pageTypes = singleton(ElementalArea::class)->supportedPageTypes();
         $count = 0;
         foreach ($pageTypes as $pageType) {
-            $pages = $pageType::get()->filter('ElementalAreaID', 0);
+            // Only pages that have the ElementalPageExtension have a known ElementalArea relation
+            if (!$this->isMigratable($pageType)) {
+                continue;
+            }
+
+            $pages = $pageType::get()->filter('Content:not', ['', null]);
+            $clearContent = $this->config()->get('clear_content');
+
+            $this->extend('updatePageFilter', $pages, $pageType);
+            $pageTypeCount = 0;
+
+            /** @var SiteTree&ElementalAreasExtension $page */
             foreach ($pages as $page) {
+                if ($this->shouldSkipMigration($page)) {
+                    continue;
+                }
+                // Fetch and clear existing content (if configured)
                 $content = $page->Content;
-                $page->Content = '';
-                // trigger area relations to be setup
-                $page->write();
-                $area = $page->ElementalArea();
-                $element = new ElementContent();
+                $pageIsLive = $page->isPublished();
+                if ($clearContent) {
+                    $page->Content = '';
+                }
+
+                // Get the area
+                $area = $this->getAreaRelationFromPage($page);
+
+                // Write the page if we're clearing content or if the area doesn't exist - we write to trigger a
+                // relationship update
+                if ($clearContent || !$area->exists()) {
+                    $page->write();
```

Comment:
I think you should do this within a try/catch, incase creating the new element with this content fails and you end up in a state where you have no content at all